### ISSUE TITLE
Are there any modin equivalents for testing and _libs pandas imports ?

### ISSUE THREAD

**Issue by PeterPirog** [modin-project/modin#4236]
Are there any modin equivalents for these pandas imports ?

```
from pandas.testing import assert_frame_equal

from pandas._libs.lib import (
    NoDefault,
    no_default,
)
from pandas._libs.missing import is_matching_na
import pandas._libs.testing as _testing
from pandas.util._exceptions import find_stack_level

from pandas.core.dtypes.common import (
    is_bool,
    is_categorical_dtype,
    is_extension_array_dtype,
    is_interval_dtype,
    is_number,
    is_numeric_dtype,
    needs_i8_conversion,
)
```

These imports are in the original pandas code and I would like to replace it.

**Comment by AntonyM71** [comment]
If we're happy to add a testing folder, for `assert_frame_equal` we could probably import the method from pandas an overwrite it's hardcoded check for a pandas (rather than median) data frame.

**Comment by vnlitvinov** [comment]
cc @modin-project/modin-core for more comments

**Comment by jbrockmendel** [comment]
find_stack_level, is_matching_na, no_default, and NoDefault you should be able to re-use directly (i.e. modin doesn't need to implement its own).  _libs.testing I discourage anyone from using directly.

**Comment by anmyachev** [comment]
@PeterPirog FYI. Modin currently implements modin.pandas.testing API (will be available in Modin 0.29.0). There are no plans to implement a private `pandas._libs` module.
